tests/asgi/test_testing_asgi.py

```diff
@@ -153,3 +153,14 @@ def test_immediate_disconnect():
 
     with pytest.raises(ConnectionError):
         client.simulate_get('/', asgi_disconnect_ttl=0)
+
+
+def test_create_scope_preserve_raw_path():
+    path_no_queries = '/cache/http%3A%2F%2Ffalconframework.org/status'
```

Comment:
It would be nice if we could reduce the amount of copy-paste in this test case by leveraging `pytest.mark.parametrize()`.